openshift/system/Dockerfile

```diff
@@ -7,17 +7,19 @@ ENV RUBY_MAJOR_VERSION=2 \
 ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}"
 ADD . ./
 ADD openshift/system/config/* ./config/
-RUN dnf -y module enable ruby:${RUBY_VERSION} nodejs:16 mysql:8.0 \
-    && dnf install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs shared-mime-info make automake gcc gcc-c++ redhat-rpm-config postgresql rubygem-irb rubygem-rdoc ruby-devel nodejs libpq-devel mysql-devel gd-devel git 'dnf-command(download)' podman-catatonit \
-    && gem install bundler --version 2.2.25 \
+RUN dnf -y --setopt=module_stream_switch=True module enable ruby:${RUBY_VERSION} nodejs:16 mysql:8.0 \
+    && dnf install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs shared-mime-info make automake gcc gcc-c++ redhat-rpm-config postgresql rubygem-irb rubygem-rdoc ruby-devel nodejs libpq-devel mysql-devel gd-devel libxml2-devel libxslt-devel git 'dnf-command(download)' podman-catatonit \
+    && BUNDLER_VERSION=$(awk '/BUNDLED WITH/ { getline; print $1 }' Gemfile.lock) \
+    && gem install --no-document bundler:$BUNDLER_VERSION \
+    && bundle config build.nokogiri --use-system-libraries \
     && bundle config set --local without development:test:licenses \
     && bundle config set --local deployment true \
     && bundle install --jobs $(grep -c processor /proc/cpuinfo) --retry=5 \
     && npm install -g yarn \
     && yarn install:safe --no-progress \
     # TODO: update to SECRET_KEY_BASE_DUMMY=1 when we upgrade to Rails 7.0
     && SECRET_KEY_BASE=rails/32947 bundle exec rake assets:precompile tmp:clear \
-    && rm -rf tmp/cache /usr/local/share/gems/cache /usr/local/share/gems/doc
+    && rm -rf node_modules /usr/local/share/gems/cache /usr/local/share/gems/doc
```

Comment:
`tmp/cache` is cleaned by `tmp:clear` 

I don't see us deleting `node_modules` elsewhere so I've put it here.